src/components/CheckBox/checkbox.module.scss

```diff
@@ -242,14 +242,237 @@
+                    }
+
+                    &.toggle {
+                        border-radius: $toggle-medium-height;
+                        height: $toggle-medium-height;
+                        width: $toggle-medium-width;
+
+                        &:after {
+                            height: $toggle-medium-after-height;
+                            left: $toggle-medium-after-left;
+                            top: $toggle-medium-after-top;
+                            width: $toggle-medium-after-width;
+                        }
+                    }
+                }
+            }
+        }
+
+        input:checked {
+            & + label {
+                .checkmark {
+                    &:after {
+                        transform: rotate(45deg) scale(0.7);
+                    }
+
+                    &.toggle {
+                        &:after {
+                            left: $toggle-medium-after-left-checked;
+                        }
+                    }
+                }
+            }
+        }
+
+        .selector-label {
+            font-size: $text-font-size-2;
+
+            &-end {
+                margin-left: $space-xxs;
+            }
+
+            &-start {
+                margin-right: $space-xxs;
+            }
+        }
+    }
+
+    &-small {
+        input {
+            & + label {
+                .checkmark {
+                    height: $checkmark-small-height;
+                    width: $checkmark-small-width;
+
+                    &:after {
+                        height: $checkmark-small-after-height;
+                        left: $checkmark-small-after-left;
+                        top: $checkmark-small-after-top;
+                        width: $checkmark-small-after-width;
+                    }
+
+                    &.toggle {
+                        border-radius: $toggle-small-height;
+                        height: $toggle-small-height;
+                        width: $toggle-small-width;
+
+                        &:after {
+                            height: $toggle-small-after-height;
+                            left: $toggle-small-after-left;
+                            top: $toggle-small-after-top;
+                            width: $toggle-small-after-width;
+                        }
+                    }
+                }
+            }
+        }
+
+        input:checked {
+            & + label {
+                .checkmark {
+                    &.toggle {
+                        &:after {
+                            left: $toggle-small-after-left-checked;
+                        }
+                    }
+                }
+            }
+        }
+
+        .selector-label {
+            font-size: $text-font-size-1;
+
+            &-end {
+                margin-left: $space-xxs;
+            }
+
+            &-start {
+                margin-right: $space-xxs;
+            }
+        }
+    }
+
+    &.disabled {
```

Comment:
this is alot of styles for disabled state. is it possible to get savings from doing a not trick on the default hovers, etc?
input:not(.disabled):hover {
  color rules
  }